src/system/LwIPEventSupport.h

```diff
@@ -0,0 +1,38 @@
+/*
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <core/CHIPError.h>
+#include <system/SystemEvent.h>
+
+namespace chip {
+namespace System {
+
+class Layer;
+class Object;
+
+class PlatformEventing
```

Comment:
any reason for using a class here?

Seeinc a class with all public static methods makes me wonder 'why not a namespace' ?

Reply:
Resolving as this was just moving code around. Still wondering why the original code was using classes instead of namespaces though.

Reply:
This has come up before; it's part of the existing coupling to the Platform layer for the LwIP case, which I haven't yet given as much attention as the socket case.